crates/bevy_scene/src/scene.rs

```diff
@@ -28,7 +28,12 @@ impl Scene {
                     })
                 }
                 for type_info in archetype.types() {
-                    if let Some(component_registration) = component_registry.get(&type_info.id()) {
+                    if let Some(component_registration) =
+                        component_registry.get(&match type_info.id() {
+                            bevy_ecs::ComponentId::RustTypeId(id) => id,
+                            _ => todo!("Handle external ids in component registry"),
```

Comment:
This may need to be fixed before merging. We have to find out how we want to manage the scene registry when it comes to external types. I don't have a full understanding of how it works currently so I didn't implement anything yet

I think I'm going to prioritize figuring out the dynamic component and systems approach before I try to fix this.